monkey/common/cloud/azure/azure_instance.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import requests
+
+LATEST_AZURE_METADATA_API_VERSION = "2019-06-04"
+AZURE_METADATA_SERVICE_URL = "http://169.254.169.254/metadata/instance?api-version=%s" % LATEST_AZURE_METADATA_API_VERSION
+
+logger = logging.getLogger(__name__)
+
+
+class AzureInstance(object):
```

Comment:
These classes are basically singletons. Why are they a class at all? What state do they keep that can't be done  when the file is loaded?

Reply:
also, we're py3.7 alredy, no need for `(object)`

Reply:
Kept them as classes to avoid changes all over, but created the all_instances file which does use them statically. 

About not using (objcet), either we change the convention all over or we don't - and IMO explicit is better than implicit 🤷‍♀️ Going to leave it as it is